pkg/operator/ceph/cluster/osd/topology/topology.go

```diff
@@ -87,22 +80,21 @@ func extractTopologyFromLabels(labels map[string]string) (map[string]string, str
 		topologyAffinity = formatTopologyAffinity(corev1.LabelZoneFailureDomain, zone)
 	}
 
-	// check for the zone k8s topology label that is GA in 1.17.
-	zone, ok = labels[corev1.LabelZoneFailureDomainStable]
-	if ok {
-		topology[zoneLabel] = zone
-		topologyAffinity = formatTopologyAffinity(corev1.LabelZoneFailureDomainStable, zone)
-	}
-
-	// ignore the region k8s topology label when it has the same value as the k8s zone label
-	if topology[regionLabel] == topology[zoneLabel] {
-		delete(topology, regionLabel)
-	}
+	kubernetesTopologyLabels := []string{"host", "zone", "region"}
```

Comment:
I'm seeing 3 loops now where there used to be 1. How can we pare this down to just 2 loops? 1 going forward and removing duplicates and the other going in reverse and setting the CRUSH topology?

What if we created a list of labels and then iterated over it like here in a quick playground prototype? IMO, this is much easier to understand than even the pre-existing code loop, which is already a bit more complicated than is easy for me to read.
https://play.golang.com/p/lIAo17gC_ex

Also, let's be sure to re-use pre-defined values like those in the k8s core api here: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/core/v1/well_known_labels.go

Can we use the topology labels (and helpers) I just noted to iterate over backwards in the next loop as well?

Reply:
Thanks updated